packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -465,7 +465,49 @@ export function appendChild(
   parentInstance: Instance,
   child: Instance | TextInstance,
 ): void {
-  parentInstance.appendChild(child);
+  const isDocumentParentNode = parentInstance.nodeType === DOCUMENT_NODE;
```

Comment:
This is changing an extremely hot path and might break a bunch of optimisations. We need to be very careful about this.